ohttp/src/rh/hpke.rs

```diff
@@ -368,6 +491,38 @@ impl HpkeR {
                 }
             };
         }
+        #[cfg(feature = "rust-hpke-pq")]
+        let context = dispatch_hpker_new! {(config, sk_r): [
+            {
+                Kem::X25519Sha256 => X25519HkdfSha256,
+                Kdf::HkdfSha256 => HkdfSha256,
+                Aead::Aes128Gcm => AesGcm128,
+                PrivateKey::X25519,
+                ReceiverContext::X25519HkdfSha256,
+                ReceiverContextX25519HkdfSha256::HkdfSha256,
+                ReceiverContextX25519HkdfSha256HkdfSha256::AesGcm128,
+            },
+            {
+                Kem::X25519Sha256 => X25519HkdfSha256,
+                Kdf::HkdfSha256 => HkdfSha256,
+                Aead::ChaCha20Poly1305 => ChaCha20Poly1305,
+                PrivateKey::X25519,
+                ReceiverContext::X25519HkdfSha256,
+                ReceiverContextX25519HkdfSha256::HkdfSha256,
+                ReceiverContextX25519HkdfSha256HkdfSha256::ChaCha20Poly1305,
+            },
+            {
+                Kem::X25519Kyber768Draft00 => X25519Kyber768Draft00,
+                Kdf::HkdfSha256 => HkdfSha256,
+                Aead::Aes128Gcm => AesGcm128,
+                PrivateKey::X25519Kyber768Draft00,
+                ReceiverContext::X25519Kyber768Draft00,
+                ReceiverContextX25519Kyber768Draft00::HkdfSha256,
+                ReceiverContextX25519Kyber768Draft00HkdfSha256::AesGcm128,
+            },
+        ]};
+
+        #[cfg(not(feature = "rust-hpke-pq"))]
```

Comment:
That trick I showed you above (where you pass down attributes into the macro), might be needed here as well.  If only to reduce the repetition.

Reply:
Works, thanks.